whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/UpdateNrtmFileRepository.java

```diff
@@ -66,6 +68,15 @@ public void saveSnapshotVersion(final NrtmVersionInfo version, final String file
         LOGGER.info("Created {} snapshot version {}", version.source().getName(), version.version());
     }
 
+    public void rotateKey(final NrtmKeyRecord newActiveKey, final NrtmKeyRecord oldActiveKey) throws JsonProcessingException {
```

Comment:
Can you add a JMX managed operation to invoke this method (including generating a new key)

Reply:
afaik UpdateNotificationFIle will be executed an we will take care of the rotation of the key
Check:
UpdateNotificationFileGenerationTestIntegration.should_add_next_signing_key-in_notification_file()
This will call nrtmKeyPairService.generateOrRotateNextKey(); which will take care if the key need to be rotated or not.

Reply:
Talked to Ed:
We need to do the rollover at ANY time...so in that case we need the JMX.
We are doing the rollover in an scheduler task right now

Reply:
Adding it in a JMX will not help in case of key compromise as spec says client needs atleast 7 days for rotation. In case of compromise we need manual intervention

Reply:
I want an easy way to trigger a rotation, without needing to update the database manually, can this be done with JMX?

Reply:
This job creates a next signing key or rotate based on RFC spec and based on specific timeline. In case of compromise and  need to rotate key a Seperate JMX call and a new function to execute should be created and not use the same task. and the we also need someway to tell client somehow  about instantly using new key  as RFC says 7 days to rotate

Reply:
I want a way to manually trigger a rollover to a new signing key, whenever we like, and I don't want to touch the database manually. For example, once this functionality goes live, we should trigger a rollover once everyone is ready.

We should also create a separate function to trigger an instant rollover without waiting for 7 days, but again we shouldn't have to touch the database manually for that either.

Reply:
I will create a two JMX call, one to rotate the next key and one to create the next key  and rotate.